R/jira.R

```diff
@@ -8,103 +8,392 @@
   jira_json_path <- file.path(folder_path,"fake_issues.json")
   jsonlite::write_json(issues,file.path(folder_path,"fake_issues.json"))
 
+  return(jira_json_path)
+}
+
+#' Create base_info_cell
+#'
+#' Creates and formats base_info_cell for fake Jira issue
+#'
+#' @param id ID of Jira issue
+#' @param self URL of API endpoint
+#' @param key key for Jira issue
+#' @param JirAgileR_id JirAgileR id
+#' @return A list named 'base_info_cell' containing all the parameters
+#' @export
+create_base_info <- function(id = 11, self = "https://project.org/jira/rest/api/latest/issue/11", key = "PROJECT-68", JirAgileR_id = 1) {
+  base_info_cell <- list(
+    id = id,
+    self = self,
+    key = key,
+    JirAgileR_id = JirAgileR_id
+  )
+
+  return(base_info_cell)
+}
+
+#' Create ext_info_cell for issues (without comments)
+#'
+#' Creates and formats ext_info_cell for fake Jira issue
+#'
+#' @return ext info cell
+#' @export
+create_ext_info_issues <- function() {
+  ext_info_cell <- list(
+    summary = list("Issue Description"),
+    issuetype = create_issuetype(),
+    components = create_components(component_details),
+    creator = create_creator(),
+    created = list("2007-07-08T06:07:06.000+0000"),
+    description = list(),
+    reporter = create_reporter(),
+    resolution = create_resolution(self_url = "https://domain.org/jira/rest/api/2/resolution/1"),
+    resolutiondate = list("2007-08-13T19:12:33.000+0000"),
+    assignee = list(),
+    updated = list("2008-05-12T08:01:39.000+0000"),
+    status = create_status()
+  )
+
+  return(ext_info_cell)
+}
+
+#' Create Issue Type
+#'
+#' Create issue type cell for fake Jira issue
+#'
+#' @param self_url URL of issue type's endpoint
+#' @param id ID of Jira issue
+#' @param description Description of issue type
+#' @param iconUrl URL of icon representing issue type
+#' @param name Name of issue type
+#' @param subtask FALSE, determines if the issue is a subtask
+#' @param avatarId ID for issue type's avatar image
+#' @return A list named 'issuetype' that represents the issue type of the Jira issue
+#' @export
+create_issuetype <- function(self_url = "https://project.org/jira/rest/api/2/issuetype/2",
+                             id = "2",
+                             description = "A new feature of the product, which has yet to be developed.",
+                             iconUrl = "https://domain.org/jira/secure/viewavatar?size=xsmall&avatarId=21141&avatarType=issuetype",
+                             name = "New Feature",
+                             subtask = FALSE,
+                             avatarId = 21141) {
+  issuetype <- list(
+    self = list(list(self_url)),
+    id = list(list(id)),
+    description = list(list(description)),
+    iconUrl = list(iconUrl),
+    name = list(name),
+    subtask = list(subtask),
+    avatarId = list(avatarId)
+  )
+
+  return(issuetype)
+}
+
+#' Create Components
+#'
+#' Creates a list populated with components and their information from components_list
```

Comment:
Same as before (I will not mention this hereafter, please assume it needs to be re-worked). For components: `\url{https://support.atlassian.com/jira-software-cloud/docs/organize-work-with-components/}`. 

Also, make sure you only refer to JIRA official documentation rather than a random blog post or website.